torch/onnx/_internal/exporter.py

```diff
@@ -78,7 +79,25 @@ def __init__(
         self.logger = logger
 
 
-class ResolvedExportOptions:
+class ResolvedExportOptions(ExportOptions):
+    """Consolidates `ExportOptions` with default values.
+    All unspecified options from `ExportOptions` are assigned a default value.
+    This is an internal class and its API may be changed at any time without notice.
+    """
+
+    # Public attributes MUST be redefined below without ``Optional[]`` from ``ExportOptions``
+    opset_version: int
+    dynamic_shapes: bool
+    op_level_debug: bool
+    logger: logging.Logger
+
+    # Private only attributes
+    decomposition_table: Dict[torch._ops.OpOverload, Callable]
+    """Dictionary with all ATen operators supported by the exporter."""
```

Comment:
docstring not accurate, more like 

`Dictionary with ATen/Prims operators and their decompositions. Exporter runs decomposition after getting the FX graph from FXGraphExtractor. If not provided, the default decomposition table is based on the full PyTorch decomposition table, excluding Prims decomposition, and any ATen operator decomposition if the ATen operator has a matching OnnxFunction in torch_lib.`

And based on the level of internal details, not sure why it needs to be exposed in this PR. Unless it is needed urgently for Prims experiments. cc @justinchuby. Ideally a much more natural api emerges after the dispatcher refactor.

Reply:
This is not publicly exposed; `ResolvedExportOptions` is internal-only. This serves as an internal extension point and it is a natural Dynamo concept

Reply:
Using docstring from https://github.com/thiagocrepaldi/pytorch/blob/c8bf6f3c699bfd9833e9b9b65a1c39878f1ad440/torch/_dynamo/eval_frame.py#L705

Reply:
This is similar to the docstring for `backend` of `DynamoOptimize` that we discussed earlier. Both need code rework to match the same concept of its original fx/dynamo counterpart, which is what stated in the docstring. I won't block on this, will revisit in dispatcher theme.

Reply:
I don't get why `"""A dictionary that maps operators to their decomposition functions."""` doesn't match the same concept.

I understand we are not giving a full description on how it is used, but it is still a truthful explanation;  we do build a decomposition_table (with whatever our registry can support) and pass it to FX APIs o consume it, as they document it they should be.

Reply:
Fair enough. I felt we are adding it before designing the purpose and the expected outcome under exporter context, which is what dispatcher design going to do anyway.

Reply:
got your point.
not all changes here are final because the rest of the code is not final.
next redesigns may refactor some of the things done on previous refactoring until everything make sense :)